### ISSUE TITLE
How to create sensible integration tests?

### ISSUE THREAD

**Issue by SirMathhman** [Vincit/objection.js#2035]
Hello all!

I've just started using Objection.js and I'm having trouble writing integration tests. It doesn't seem clear to me in the documentation how to do so, or maybe I'm just blind!

The part that has been really ruffling my feathers for a bit now is the inability to override `tableName`, either the field or the property getter for it.

I have three tables:

1) A production table, obviously
2) A development table
3) An integration testing table that is, more or less, created when the tests begin and is immediately torn down when they are completed

The production and development tables have the same name, but the integration testing table has a different name, e.g. the development table has a name of 'Quote' and the integration testing table has a name of 'TestQuote'. I do this because sometimes I would like to test persistent data across development sessions (i.e. running the program manually) and data across automated tests, which should only exist when the tests are running. But because the data is modeled similarly, the logical process would be to have a shared model/hierarchy and have those two tables extend that superclass with the desired table name, the DRY principle... 

For example:
```
export class AbstractQuote extends Model {
   // ... stuff here

  // Is this meaningful, if it's just going to be overridden?
  static get tableName() {
    return "test"  
  }  
}

export class ActualQuote extends Model {

  // override tableName?
  static get tableName(){
    return "Quote" 
  }
}

export class TestQuote extends Model {
  static get tableName(){
    return "TestQuote" 
  }
}
```

However, according to my understanding of JavaScript/TypeScript, it's not possible override static methods (beats the point of the 'static' keyword). Consequently, I am inquiring about:

1) What solutions/workarounds are there for this issue?
    1.1) How have you approached this problem?
2) Is it meaningful to build a hierarchy of tables?
3) Is this even the best way to implement DRY, or are there other methods?

TLDR: I'm having trouble with building a hierarchy of tables.

**Comment by koskimas** [comment]
It makes much more sense to have a separate database for each environment, and use the same table names.

**Comment by SirMathhman** [comment]
What do you mean by a "dynamically changing table?"

**Comment by koskimas** [comment]
I mean the changing table name. If the target table's name can be different in different situations, there's no easy way to build the relation mappings.

I'd just use the different db per environment case. Otherwise you'll have a really confusing codebase.
